### PR TITLE
Refactor CheckBoxWithStickyDiv story in CheckBox using Hooks

### PR DESCRIPTION
**Pull request by nolleto**
#### What does this PR do?

Refactor `src/js/components/CheckBox/stories/WithStickyDiv.js` using Hooks

#### Where should the reviewer start?

`src/js/components/CheckBox/stories/WithStickyDiv.js`

#### What testing has been done on this PR?

Running `yarn test` and `yarn storybook` and tested the component behavior.

#### How should this be manually tested?

Running `yarn test`

#### Any background context you want to provide?

I've changed the state name `checked` to `checks` because I believe is a better name since this represents an array. Also, I've done another refactors.

#### What are the relevant issues?

#3394 

#### Screenshots (if appropriate)

N/A

#### Do the grommet docs need to be updated?

 No

#### Should this PR be mentioned in the release notes?

No

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible

### CODE REVIEW

File: src/js/components/CheckBox/stories/WithStickyDiv.js [hunk 1]
```diff
@@ -1,61 +1,58 @@
...
-            </Box>
-            {checkboxes.map(item => (
-              <CheckBox
-                key={item}
-                checked={checked.indexOf(item) !== -1}
-                label={item}
-                onChange={e => this.onCheck(e, item)}
-              />
-            ))}
+const getChecks = ({ target }, checked, value) => {
```
Comment (oorestisime) [inline src/js/components/CheckBox/stories/WithStickyDiv.js]: can we add those functions inside the story. maybe in a normal code you want this separated but lets help the users see and get the code they need easier in storybook. plus i don't see the reason of creating a function to get checks. if you leave it inside the component then you can do that directly.

File: src/js/components/CheckBox/stories/WithStickyDiv.js [hunk 2]
```diff
@@ -1,61 +1,53 @@
-import React, { Component } from 'react';
-import { storiesOf } from '@storybook/react';
+import { Box, CheckBox, Grommet } from 'grommet';
+import React, { useState } from 'react';
 
-import { Box, Grommet, CheckBox } from 'grommet';
 import { grommet } from 'grommet/themes';
+import { storiesOf } from '@storybook/react';
```
Comment (ShimiSun) [inline src/js/components/CheckBox/stories/WithStickyDiv.js]: Can we please keep the order of imports as it was?

Comment (nolleto) [reply src/js/components/CheckBox/stories/WithStickyDiv.js]: Sure

File: src/js/components/CheckBox/stories/WithStickyDiv.js [hunk 3]
```diff
@@ -1,61 +1,53 @@
...
-  };
+const boxStyle = {
+  position: 'relative',
+  display: 'block',
+};
+const titleBoxBackground = { color: 'neutral-1' };
+const titleBoxStyle = { position: 'sticky', top: 0 };
+
+const checkboxes = Array(8)
+  .fill()
```
Comment (ShimiSun) [inline src/js/components/CheckBox/stories/WithStickyDiv.js]: Thank you for pulling `checkboxes` out of the state, makes much more sense.

File: src/js/components/CheckBox/stories/WithStickyDiv.js [hunk 4]
```diff
@@ -1,61 +1,53 @@
...
+    <Grommet theme={grommet}>
+      <Box pad="large" align="center">
+        <Box height="120px" width="120px" overflow="auto" style={boxStyle}>
+          <Box background={titleBoxBackground} style={titleBoxStyle}>
+            Click &amp; Scroll
           </Box>
+          {checkboxes.map(item => (
+            <CheckBox
+              key={item}
+              checked={checks.includes(item)}
```
Comment (ShimiSun) [inline src/js/components/CheckBox/stories/WithStickyDiv.js]: Thanks for cleaning up using `includes`

**Comment by ShimiSun** [comment]
Verified with https://www.chromaticqa.com/build?appId=5d7a63beda813e00202faf9d&number=71

**Comment by ShimiSun** [comment]
Nice work! thank you for your contribution!

**Pull request merged**